### PR TITLE
Update CDx related entities and add script to import initial CDx data

### PR DESCRIPTION
**Pull request by calvinlu3**
This fixes https://github.com/oncokb/oncokb-transcript/issues/105 and https://github.com/oncokb/oncokb-pipeline/issues/114

Instead of curating on the FdaSubmission page, we will curate on the CDx page.
image

### CODE REVIEW

File: jhipster-jdl.jdl [hunk 1]
```diff
@@ -225,21 +237,22 @@ relationship OneToMany {
...
     Drug{brands} to DrugBrand
 }
 
 relationship ManyToMany {
     Alteration to Gene
     CompanionDiagnosticDevice to SpecimenType
     ClinicalTrialsGovCondition to CancerType
+    DeviceUsageIndication to Alteration
+    DeviceUsageIndication to Drug
+    DeviceUsageIndication to FdaSubmission
```
Comment (calvinlu3) [inline jhipster-jdl.jdl:252]: Updated DeviceUsageIndication relationships
- Drugs: allow for combination druugs
- FdaSubmission: allow for multiple PMAs where evidence is retrieved from
- Alteration: multiple alterations in association

File: src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react';
```
Comment (calvinlu3) [inline src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx:1]: Deleted `FdaSubmissionPanel.tsx` and copied contents to this file.

File: src/main/webapp/app/shared/modal/SimpleConfirmModal.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  onCancel?: () => void;
+  onConfirm?: () => void;
+}> = props => {
+  const onCancel = (event?: any) => {
+    if (event) event.preventDefault();
+    if (props.onCancel) props.onCancel();
+  };
+  return (
+    <Modal isOpen={props.show} toggle={() => onCancel()}>
+      <ModalHeader toggle={() => onCancel()}>{props.title ? props.title : 'Please confirm'}</ModalHeader>
```
Comment (zhx828) [inline src/main/webapp/app/shared/modal/SimpleConfirmModal.tsx:17]:
```suggestion
      <ModalHeader toggle={() => onCancel()}>{props.title || 'Please confirm'}</ModalHeader>
```

File: src/main/webapp/app/shared/select/FdaSubmissionSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        setFdaSubmissionList(options);
+      }
+    };
+    loadFdaSubmissionOptions(cdxId);
+    if (!cdxId) {
+      setFdaSubmissionValue(null);
+    }
+  }, [cdxId]);
+
+  const onFdaSubmissionChangee = (option, actionMeta) => {
```
Comment (zhx828) [inline src/main/webapp/app/shared/select/FdaSubmissionSelect.tsx:37]:
```suggestion
  const onFdaSubmissionChange = (option, actionMeta) => {
```

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+import CompactButton from '../button/CompactButton';
+import { SimpleConfirmModal } from '../modal/SimpleConfirmModal';
+
+interface CdxBiomarkerAssociationTableProps extends StoreProps {
+  editable?: boolean;
+  companionDiagnosticDeviceId: number;
+}
+
+export const CdxBiomarkerAssociationTable: React.FunctionComponent<CdxBiomarkerAssociationTableProps> = props => {
+  const [indications, setIndications] = useState([]);
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx:19]: You have `association` and `indication` concepts and they are the same I suppose. Could you consolidate?

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+  const [showModal, setShowModal] = useState(false);
+  const [indicationToDeleteId, setIndicationToDeleteId] = useState(null);
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx:21]: I don't think we need this state.

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            {original.drugs &&
+              original.drugs
+                .map(drug => drug.name)
+                .sort()
+                .join(' + ')}
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx:108]: Should we use `getTreatmentName` in the utils?

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            <CompactButton
+              text="Delete"
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx:150]: Do we need to add the tooltip? I think the trash icon is very self-explanatory .

File: src/main/webapp/app/shared/table/OncoKBTable.scss [hunk 9]
```diff
@@ -1,9 +1,53 @@
 .oncokb-table {
   word-wrap: break-word;
+  border: 1px solid rgba(0, 0, 0, 0.1);
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/OncoKBTable.scss:3]: why we do `rgba` here?

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            {original.alterations &&
+              original.alterations
+                .map(alteration => alteration.name)
```
Comment (zhx828) [inline src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx:72]: Use `getAlterationName`?

File: src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx (hunk 6 above)
Comment (calvinlu3) [reply src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx]: It is to figure out which row we are deleting from the table. It is set when the delete button is clicked, so when we confirm in the the modal, we know which to delete.

File: src/main/webapp/app/shared/table/OncoKBTable.scss (hunk 9 above)
Comment (calvinlu3) [reply src/main/webapp/app/shared/table/OncoKBTable.scss:3]: I'm trying to get a different shade of gray. Mostly copied the styles from public website.

File: src/main/webapp/app/layout.tsx
Comment (calvinlu3) [inline src/main/webapp/app/layout.tsx:1]: Added a listener to `history` so that the curation panel shows up when on a certain page

**Comment by calvinlu3** [comment]
- Renamed `DeviceUsageIndication` to `BiomarkerAssociation`
- Refresh biomarker table after add/delete
- Remove CompactButton and use regular button
- Change button to small

File: src/main/resources/config/liquibase/data/fda_submission.csv [hunk 11]
```diff
@@ -1,10 +1,10 @@
 id|number|supplement_number|device_name|generic_name|date_received|decision_date|description|curated|genetic|companion_diagnostic_device_id|type_id
-1|P140020|S020|BRACAnalysis CDx|cancer-related germline gene mutation detection system|2019-12-06 00:00:00|2020-05-19 00:00:00|Approval to expand the intended use of BRACAnalysis® CDx to include a companion diagnostic indication for BRCA1/2 mutations in patients with metastatic castration resistant prostate cancer who may benefit from treatment with Lynparza® (olaparib).|FALSE|TRUE|1|1
```
Comment (zhx828) [inline src/main/resources/config/liquibase/data/fda_submission.csv:2]: I'm curious on how this file is generated and why all the trademarks have been removed.

File: src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+      .createBiomarkerAssociation(biomarkerAssociationDTO)
+      .then(() => {
+        notifySuccess('Biomarker association added.');
+        props.getBiomarkerAssociations(id);
+      })
+      .catch(error => notifyError(error));
+  };
+
+  const redirectToCreateAlteration = () => {
+    history.push('/alteration/new');
```
Comment (zhx828) [inline src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx:60]: Do we have ROUTE constant that manages all these strings?

**Review by zhx828 — CHANGES_REQUESTED** [review]
LGTM

File: src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx (hunk 12 above)
Comment (calvinlu3) [reply src/main/webapp/app/components/curationPanel/CompanionDiagnosticDevicePanel.tsx]: Updated to use the util function that generates the route based on the entity type and action.

File: src/main/resources/config/liquibase/data/fda_submission.csv (hunk 11 above)
Comment (calvinlu3) [reply src/main/resources/config/liquibase/data/fda_submission.csv:2]: This will be restored in this ticket: https://github.com/oncokb/oncokb/issues/3520

**Pull request merged**